mesonbuild/linkers/linkers.py

```diff
@@ -174,17 +174,26 @@ def __init__(self, exelist: T.List[str]):
         self.id = 'ar'
         pc, stdo = mesonlib.Popen_safe(self.exelist + ['-h'])[0:2]
         # Enable deterministic builds if they are available.
+        stdargs = 'csr'
+        thinargs = ''
         if '[D]' in stdo:
-            self.std_args = ['csrD']
-        else:
-            self.std_args = ['csr']
+            stdargs += 'D'
+        if '[T]':
+            thinargs = 'T'
+        self.std_args = [stdargs]
+        self.std_thin_args = [stdargs + thinargs]
         self.can_rsp = '@<' in stdo
 
     def can_linker_accept_rsp(self) -> bool:
         return self.can_rsp
 
-    def get_std_link_args(self) -> T.List[str]:
-        return self.std_args
+    def get_std_link_args(self, is_thin: bool) -> T.List[str]:
+        # not supported by armar
+        # FIXME: osx ld rejects this: "file built for unknown-unsupported file format"
```

Comment:
I don't understand this OSX failure.

From https://github.com/eli-schwartz/meson/runs/3693121858

```
cc  -o prog prog.p/prog.c.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error libhelper.a
ld: warning: ignoring file libhelper.a, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x74 0x68 0x69 0x6E 0x3E 0x0A 0x2F 0x20 0x20 0x20 0x20 0x20 0x20 0x20 )
Undefined symbols for architecture x86_64:
  "_func", referenced from:
      _main in prog.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```